qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java

```diff
@@ -1091,6 +1091,13 @@ private String loadInfoDocument(String type) throws IOException {
         return m.group(1);
     }
 
+    @Override
+    protected boolean getStrictDeprecationMode() {
+        // Certain APIs like 'update' changed paths between 6.x and 7.x. To allow these tests to run, we
+        // use the deprecated path and disable strict deprecation checks.
+        return false;
+    }
+
```

Comment:
I found I had to do this too and didn't like it. EsRestTestCase has a "fail or nothing" approach to testing responses. 

Either the whole test fails (strictDeprecationMode is too strict) or in the alternative lax mode we blindly accept _any_ warnings as ok - we can't check we got the expected warnings because ESRestTestCase does not work with the `assertWarnings` feature declared in base class ESTestCase. I'm going to look into adding that support to ESRestTestCase

Reply:
👍 thanks for this work, I'll avoid using `getStrictDeprecationMode`.